aws-sns-subscription/src/main/java/software/amazon/sns/subscription/CreateHandler.java

```diff
@@ -33,52 +21,24 @@ protected ProgressEvent<ResourceModel, CallbackContext> handleRequest(
-                    .then(progress ->
-                        proxy.initiate("AWS-SNS-Subscription::Create", proxyClient, model, callbackContext)
-                        .translateToServiceRequest(Translator::translateToCreateRequest)
-                        .makeServiceCall((subscribeRequest, client) -> createSubscription(subscribeRequest, client, model))
-                        .progress())
-                    .then(progress -> new ReadHandler().handleRequest(proxy, request, callbackContext, proxyClient, logger));
-
-    }
-
-    private SubscribeResponse createSubscription(
-        final SubscribeRequest subscribeRequest,
-        final ProxyClient<SnsClient> proxyClient,
-        final ResourceModel model)  {
-
-        final SubscribeResponse subscribeResponse;
-
-        // exception thrown if topic not found
-        retrieveTopicAttributes(Translator.translateToCheckTopicRequest(model), proxyClient);
-
-        try {
-            subscribeResponse = proxyClient.injectCredentialsAndInvokeV2(subscribeRequest, proxyClient.client()::subscribe);
-            model.setSubscriptionArn(subscribeResponse.subscriptionArn());
-        } catch (final SubscriptionLimitExceededException e) {
-            throw new CfnServiceLimitExceededException(e);
-        } catch (final FilterPolicyLimitExceededException e) {
-            throw new CfnServiceLimitExceededException(e);
-        } catch (final InvalidParameterException e) {
-            throw new CfnInvalidRequestException(e);
-        } catch (final InternalErrorException e) {
-            throw new CfnInternalFailureException(e);
-        } catch (final NotFoundException e) {
-            throw new CfnNotFoundException(e);
-        } catch (final AuthorizationErrorException e) {
-            throw new CfnAccessDeniedException(e);
-        } catch (final InvalidSecurityException e) {
-            throw new CfnInvalidCredentialsException(e);
-        }  catch (final Exception e) {
-            throw new CfnInternalFailureException(e);
+        if (resourceModel == null || StringUtils.isNullOrEmpty(resourceModel.getProtocol())) {
+            return ProgressEvent.failed(resourceModel, callbackContext, HandlerErrorCode.InvalidRequest, "Protocol is required");
+        } else if (StringUtils.isNullOrEmpty(resourceModel.getTopicArn())) {
+            return ProgressEvent.failed(resourceModel, callbackContext, HandlerErrorCode.InvalidRequest, "Topic ARN is required");
         }
 
-        logger.log(String.format("%s successfully created.", ResourceModel.TYPE_NAME));
-        return subscribeResponse;
+        logger.log(String.format("[StackId: %s, ClientRequestToken: %s] Calling Create SNS Subscription", request.getStackId(),
+                request.getClientRequestToken()));
+
+        return proxy.initiate("AWS-SNS-Subscription::Create", proxyClient, resourceModel, callbackContext)
+                    .translateToServiceRequest(Translator::translateToCreateRequest)
```

Comment:
do we need to handle this if the region is null?

https://code.amazon.com/packages/AWS21Workflows/blobs/mainline/--/src/com/amazon/aws21/workflows/resources/sns/subscription/SubscriptionActivitiesCustomImpl.java#:~:text=if%20(subscription,())%2C%20true)%3B